Allow to post path strings to an existing image in image fields

snehalbaghel opened the pull request:
Fedora Badges keeps a separate repo of badge images. It would make the migration a lot easier if badgr could accept path strings to existing images so we can mount our repo in the MEDIA_ROOT and point to those images.

The paths can be sent in the following **format**: `file:///uploads/issuer/xyz.png`
The above will point to: `MEDIA_ROOT/uploads/issuer/xyz.png`

Any path that point's outside the media folder will fail. This can be enabled by **ALLOW_IMAGE_PATHS = True** in settings.

If this PR is acceptable let me know so I can write tests for it. Further if there is a better way to do this I would like to know.

ottonomy commented:
@snehalbaghel thanks for the pull request. This seems like an interesting use case. I appreciate the fact that it is off by default. A couple questions to consider as we review:

* Is the path the right format of data to accept so that implementers will encounter few problems when they attempt to use this option with a variety of django-storages backends? It seems reasonable, but URL might be another reasonable choice.
* Are there any notes that should be added to README about this option?
* Are there security items that should be noted? This would allow images to be placed or replaced in media storage without going through the script scrubbing code.
* Are there relevant unit tests that should be added? Given that this functionality won't be used by our team internally, some tests would make sure that it doesn't break in subsequent releases.

snehalbaghel commented:
Hey @ottonomy, I've updated the PR to use default_storage so the feature should work with paths with other backends as well. I've also added some notes and tests according to your suggestions. Let me know if you have any other suggestions.

snehalbaghel commented:
@ottonomy can you review this?
